eng/common/native/init-compiler.sh

```diff
@@ -58,10 +60,25 @@ check_version_exists() {
     echo "$desired_version"
 }
 
+set_compiler_version_from_CC() {
+    local patchVersion
+    version="$("$CC" -dumpversion)"
+    if [ "$version" = "" ]; then
+        echo "$CC -dumpversion didn't provide a version"
+        exit 1
+    fi
+    # gcc and clang often display 3 part versions. However, gcc can show only 1 part in some environments.
+    IFS=. read -r majorVersion minorVersion patchVersion <<EOF
+$version
+EOF
+    unset patchVersion
```

Comment:
nit, can use discard (to avoid `unset`):
```suggestion
        IFS=. read -r majorVersion minorVersion _ <<EOF
$version
EOF
```

Reply:
Fixed. I thought `$_` was a reserved variable in shell, but POSIX leaves it undefined.